src/main/java/teammates/logic/core/StudentsLogic.java

```diff
@@ -628,16 +620,24 @@ private StudentEnrollDetails enrollStudent(StudentAttributes validStudentAttribu
         return studentList;
     }
 
-    private List<String> getInvalidityInfoInDuplicatedEmail(String email,
-            List<String> studentEmailList, String[] linesArray) {
-        List<String> info = new ArrayList<>();
-        info.add("Same email address as the student in line \"" + linesArray[studentEmailList.indexOf(email) + 1] + "\"");
-        return info;
+    private String getInvalidityInfoInDuplicatedEmail(String email,
+            List<StudentAttributes> studentList, String[] linesArray) {
+        for (int index = 0; index < studentList.size(); index++) {
+            if (studentList.get(index).email.equalsIgnoreCase(email)) {
+                return "Same email address as the student in line \"" + linesArray[index + 1] + "\"";
+            }
+        }
+        return "";
     }
 
-    private boolean isStudentEmailDuplicated(String email,
-            List<String> studentEmailList) {
-        return studentEmailList.contains(email);
+    private int getDuplicateIndex(String email,
+            List<StudentAttributes> studentList) {
+        for (int index = 0; index < studentList.size(); index++) {
+            if (studentList.get(index).email.equalsIgnoreCase(email)) {
```

Comment:
is there a reason for using `equalsIgnoreCase`? The previous code was something like `linesArray[studentEmailList.indexOf(email) + 1]`, and the method `indexOf` uses the `equals` method for comparison if I'm not wrong ^^